File: problems/1945-sum-of-digits-of-string-after-convert/solution.java
class Solution {
    public int getLucky(String s, int k) {
        StringBuilder sb = new StringBuilder();
        for (char c : s.toCharArray()) {
            sb.append(c - 'a' + 1);
        }
        String converted = sb.toString();
        int num = 0;
        for (int i = 0; i < converted.length(); i++) {
            num += converted.charAt(i) - '0';
        }
        for (int i = 1; i < k; i++) {
            int sum = 0;
            String numStr = String.valueOf(num);
            for (int j = 0; j < numStr.length(); j++) {
                sum += numStr.charAt(j) - '0';
            }
            num = sum;
        }
        return num;
    }
}

File: problems/1945-sum-of-digits-of-string-after-convert/solution.c
int getLucky(char* s, int k) {
    int num = 0;
    for (int i = 0; s[i] != '\0'; i++) {
        int val = s[i] - 'a' + 1;
        if (val < 10) {
            num += val;
        } else {
            num += (val / 10) + (val % 10);
        }
    }

    for (int i = 1; i < k; i++) {
        int next_num = 0;
        while (num > 0) {
            next_num += num % 10;
            num /= 10;
        }
        num = next_num;
    }
    return num;
}